packages/components/src/components/ColorPicker/ColorValueInput.tsx

```diff
@@ -0,0 +1,177 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { type ForwardedRef, forwardRef, useEffect, useState } from 'react';
+
+import { Select } from '../Select/Select';
+import { TextInput } from '../TextInput/TextInput';
+
+import styles from './styles/customColorPicker.module.scss';
+import { type RgbaColor } from './types';
+import {
+    DEFAULT_COLOR,
+    DEFAULT_FORMAT,
+    getLimitedColorChannelValue,
+    hexColorToRgba,
+    isValidHexColor,
+    rgbColorToHex,
+} from './utils';
+
+type ColorFormat = 'HEX' | 'RGBA';
+
+type ColorValueInputProps = {
+    /**
+     * @ignore
+     * The active color in the color picker, passed down from the root component
+     */
+    currentColor?: RgbaColor;
+    /**
+     * @ignore
+     * Event handler called when the color changes, passed down from the root component
+     */
+    onColorChange?: (color: RgbaColor) => void;
+    /**
+     * @ignore
+     * The format to use for the color input, passed down from the root component
+     */
+    currentFormat?: ColorFormat;
+    /**
+     * @ignore
+     * The format to use for the color input, passed down from the root component
+     */
+    setCurrentFormat?: (format: ColorFormat) => void;
+    /**
+     * The test id of the color picker value input
+     */
+    'data-test-id'?: string;
+};
+
+export const ColorValueInput = (
+    {
+        currentColor = DEFAULT_COLOR,
+        onColorChange = () => {},
+        currentFormat = DEFAULT_FORMAT,
+        setCurrentFormat = () => {},
+        'data-test-id': dataTestId = 'color-picker-value-input',
+    }: ColorValueInputProps,
+    forwardedRef: ForwardedRef<HTMLDivElement>,
+) => {
+    const [hexColorValue, setHexColorValue] = useState<string>(rgbColorToHex(currentColor));
+
+    useEffect(() => {
+        setHexColorValue(rgbColorToHex(currentColor));
+    }, [currentColor]);
+
+    return (
+        <div className={styles.inputs} data-test-id={dataTestId} ref={forwardedRef}>
+            <div className={styles.colorFormatInput}>
+                <Select
+                    aria-label="Select an item"
```

Comment:
if multiple of these are on the page using the same `aria-label` could be problematic, this should probably come from a prop

Reply:
are you sure? i feel like using a screen reader on a form with a few of this would yield many readouts of the same text. Also, it might be worth considering adding `aria-labelledby` if a dev wants to pair this with his own label